src/hotspot/share/services/memJfrReporter.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "jfr/jfrEvents.hpp"
+#include "services/memJfrReporter.hpp"
+#include "services/memTracker.hpp"
+#include "services/nmtUsage.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/ticks.hpp"
+
+Ticks MemJFRCurrentUsage::_timestamp;
+NMTUsage* MemJFRCurrentUsage::_usage = nullptr;
+
+
+NMTUsage* MemJFRCurrentUsage::get_usage() {
+  Tickspan since_baselined = Ticks::now() - _timestamp;
+
+  if (_usage == nullptr) {
+    // First time, create a new NMTUsage.
+    _usage = new NMTUsage(NMTUsage::OptionsNoTS);
+  } else if (since_baselined.milliseconds() < AgeThreshold) {
+    // There is recent enough usage information, return it.
+    return _usage;
+  }
+
+  // Refresh the usage information.
+  _usage->refresh();
+  _timestamp.stamp();
+
+  return _usage;
+}
+
+Ticks MemJFRCurrentUsage::get_timestamp() {
+  return _timestamp;
+}
+
+void MemJFRReporter::send_total_event() {
+  if (!MemTracker::enabled()) {
+    return;
+  }
+
+  NMTUsage* usage = MemJFRCurrentUsage::get_usage();
+  Ticks timestamp = MemJFRCurrentUsage::get_timestamp();
+
+  EventNativeMemoryUsageTotal event;
```

Comment:
Since you are explicitly setting the startTime field yourself, you can pass 'UNTIMED' to the event constructor. This will bypass the automatic setting of _starttime as part of the constructor.

EventNativeMemoryUsageTotal event(UNTIMED);